### ISSUE TITLE
[Select] Improve accessibility of the clearable select

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#2844]
### Provide a general summary of the issue here

Right now, the 'clear' button in the `clearable` select component is not focusable. It is also not explained how to use it for the assistive technology users, because it's not a typical behaviour for select (in the native select we have empty option). The keyboard user doesn't know that the Backspace/Delete key can be used to remove the selected option (this is not described in any way).

Maybe we can add `aria-describedby` with the description, like we did for the removable tags or a tooltip with the explanation, but a better solution is to have `clear` button focusable.

Please also check and update documentation (when needed).

This issue can be connected with #2630.  

### 👤 Your name

Anna Sobczak

### 🧢 Your Product/Team

SLDS
